mlflow/models/flavor_backend_registry.py

```diff
@@ -0,0 +1,25 @@
+"""
+Registry of supported flavor backends. Contains a mapping of flavors to flavor backends. This
+mapping is used to select suitable flavor when deploying generic MLflow models.
+
+Flavor backend can deploy particular flavor locally to generate predictions or to deploy as a local
+REST api endpoint. Not all flavors have a flavor backend.
+"""
+import mlflow.pyfunc as pyfunc
+from mlflow.pyfunc.backend import PyFuncBackend
+from mlflow.rfunc.backend import RFuncBackend
+
+
+_flavor_backends = {
+    pyfunc.FLAVOR_NAME: PyFuncBackend,
+    "crate": RFuncBackend
+}
+
+
+def get_flavor_backend(model, **kwargs):
+    for flavor_name, flavor_config in model.flavors.items():
```

Comment:
Should we sort this somehow so that it's deterministic? I think in Python 3 dicts are ordered but in Python 2 they aren't.

Reply:
We could also log backend that's ultimately selected here for visibility to the user

Reply:
1. Good question. I chose to let the model to define the order, at least in python3. It is non-deterministic in python2, but python2 is being phased out and this should only matter in case where there is multiple available flavor backends. The alternative is to define a flavor order in mlflow. That would be deterministic for both python 2 and 3 but would also give less flexibility to custom flavor authors. I would also be worried that we would need to keep that order in future versions of mlflow. I think I would prefer to keep it as is and if there is a need to have deterministic flavor selection we can add an optional flavor argument specifying which flavor should be used.

2. That is a good point. I'll add eprint.

Reply:
Got it, yeah having the model specify the order makes sense, as does having users pass an optional flavor argument for determinism. If we really needed to we could make an additive change post-1.0 to allow the user to specify a default flavor ordering via an additional yaml field (although I guess 1.0 would not honor this ordering), but I think explicitly passing the flavor should address most use cases.